src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -204,6 +225,110 @@ send_exception_message(const char* msg)
 
 }
 
+//////////////////////////////////////////////////////////////
+// xrt_graph C++ API implementations (xrt_graph.h)
+//////////////////////////////////////////////////////////////
+namespace xrt {
+
+graph::
+graph(const xrt::device& device, const xrt::uuid& xclbin_id, const std::string& name)
+  : handle(open_graph(device, xclbin_id, name))
+{}
+
+void
+graph::
+reset() const
+{
+  handle->reset();
+}
+
+uint64_t
+graph::
+get_timestamp() const
+{
+  return (handle->get_timestamp());
+}
+
+void
+graph::
+run()
+{
+  handle->run(0);
+}
+
+void
+graph::
+run(uint32_t iterations)
+{
+  handle->run(iterations);
+}
+
+void
+graph::
+wait_done(int timeout_ms)
```

Comment:
`wait_done` sounds a little bit strange.

Reply:
Furthermore I am a little worried by `_ms` here  that does not smell very C++-ish.
Could we use something similar to https://en.cppreference.com/w/cpp/thread/sleep_for instead?

Reply:
Yes,  should use std::chrono::milliseconds, but do keep wait(unsinged int timeout_ms) as inline (see xrt_kernel.h).  I actually thing the timeout_ms is simpler than chrono, but agree we should have the chrono one also

Reply:
I put this onto my todo list to add std::chrono::milliseconds interface. Will implement it in next PR.